src/phoenix/server/api/routers/v1/dataset_examples.py

```diff
@@ -0,0 +1,72 @@
+from sqlalchemy import func, select
+from starlette.responses import JSONResponse, Response
+from starlette.status import HTTP_404_NOT_FOUND
+from strawberry.relay import GlobalID
+
+from phoenix.db.models import DatasetExample, DatasetExampleRevision, DatasetVersion
+
+
+async def list_dataset_examples(request):
+    dataset_id = GlobalID.from_id(request.path_params["id"])
+    raw_version_id = request.query_params.get("version")
+    version_id = GlobalID.from_id(raw_version_id) if raw_version_id else None
+
+    async with request.app.state.db() as session:
+        if (type_name := dataset_id.type_name) != "Dataset":
+            return Response(
+                content=f"ID {dataset_id} refers to a f{type_name}", status_code=HTTP_404_NOT_FOUND
+            )
```

Comment:
good point, though we should probably note it's not as expensive as it seems because sqlalchemy is doing the connection pooling (the connection is already made)